cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -234,14 +231,9 @@ func (sub EthereumSub) Start(completionEvent *sync.WaitGroup) {
 			// The user who starts this must provide a valid last processed block
 			if lastProcessedBlock.Cmp(big.NewInt(0)) == 0 {
 				lastProcessedBlock.Sub(endingBlock, big.NewInt(1))
-			} else if catchUpNeeded {
-				// if we need to catch up, then do that and let the relayer know that we don't need to catch up again
-				catchUpNeeded = false
-			} else {
-				lastProcessedBlock = endingBlock
```

Comment:
lastProcessedBlock only updated if the block processed by relayer successfully. we can't set it as as endingBlock everytime if we should handling errors with blocks as https://github.com/Sifchain/sifnode/issues/670.